File: backend/src/main/java/ec/edu/istr/violentometro/model/UserAnswer.java
package ec.edu.istr.violentometro.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.ColumnDefault;

import java.time.OffsetDateTime;

@Entity
@Table(name = "user_answer")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class UserAnswer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_answer", nullable = false)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_app_user")
    private AppUser idAppUser;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_question")
    private Question idQuestion;

    @Column(name = "answer")
    private Boolean answer;

    @ColumnDefault("now()")
    @Column(name = "created_at")
    private OffsetDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = OffsetDateTime.now();
    }


}

File: frontend/src/app/models/user-answer.ts
export interface UserAnswerRequest {
    idAppUser: number;
    idQuestion: number;
    answer: boolean;
}

export interface UserAnswerResponse {
    id: number;
    userId: number;
    questionId: number;
    answer: boolean;
    createdAt: string;
}

export interface UserAnswerDTO {
    idAnswer:   number;
    answer:     boolean;
    idAppUser:  number;
    idQuestion: number;
}